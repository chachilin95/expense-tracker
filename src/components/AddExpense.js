import React from 'react';

export default class AddExpense extends React.Component {
    render() {
        return (
            <div>
                This is the Add Expense Page
            </div>
        );
    }
}