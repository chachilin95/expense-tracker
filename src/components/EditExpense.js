import React from 'react';

export default class EditExpense extends React.Component {
    render() {
        return (
            <div>
                This is the Edit Expense Page
            </div>
        );
    }
}