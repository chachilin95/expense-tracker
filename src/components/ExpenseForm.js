import React from 'react';

export default class ExpenseForm extends React.Component {
    state = {
        description: '',
        amount: 0.00,
        note: ''
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };

    onAmountChange = (e) => {
        const dollarFormat = /^\d+(\.\d{0,2})?$/;
        const amount = e.target.value;

        if (amount.match(dollarFormat)) {
            this.setState(() => ({ amount }));
        }
    };

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };

    render() {
        return (
            <div>
                <form>
                    <input 
                        type='text'
                        placeholder='Description'
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}/>
                    <input 
                        type='number'
                        placeholder='Amount'
                        value={this.state.amount}
                        onChange={this.onAmountChange}/>
                    <textarea
                        placeholder='Add a note... (optional)'
                        value={this.state.note}
                        onChange={this.onNoteChange}/>
                    <button>Add Expense</button>
                </form>
            </div>
        );
    }
}