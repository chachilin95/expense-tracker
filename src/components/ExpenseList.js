import React from 'react';
import { connect } from 'react-redux';

const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        <p>filters: {props.filters.text}</p>
        <p>expenses: {props.expenses.length}</p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseList);