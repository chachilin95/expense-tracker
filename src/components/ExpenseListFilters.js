import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import uuid from 'uuid';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null,
        startDateId: uuid(),
        endDateId: uuid()
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    onSortChange = (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate();
        } else if (e.target.value === 'amount') {
            this.props.sortByAmount();
        }
    };

    render() {
        return (
            <div className='content-container'>
                <div className='input-group'>
                    <div className='input-group__item'>
                        <input
                            className='text-input'
                            onChange={this.onTextChange}
                            placeholder='Search expenses'
                            type='text'
                            value={this.props.filters.text}/>
                    </div>
                    <div className='input-group__item'>
                        <select
                            className='select'
                            onChange={this.onSortChange}
                            value={this.props.filters.sortBy}>
                            <option value='date'>Date</option>
                            <option value='amount'>Amount</option>
                        </select>
                    </div>
                    <div className='input-group__item'>
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            startDateId={this.state.startDateId}
                            endDate={this.props.filters.endDate}
                            endDateId={this.state.endDateId}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                            showClearDates />
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (date) => dispatch(setStartDate(date)),
    setEndDate: (date) => dispatch(setEndDate(date))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);