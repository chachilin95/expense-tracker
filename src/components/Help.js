import React from 'react';

export default class Help extends React.Component {
    render() {
        return (
            <div>
                This is the Help Page
            </div>
        );
    }
}