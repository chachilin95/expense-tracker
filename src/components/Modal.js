import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#app'); // added due to an API change for v3.1.7

const OptionModal = (props) => (
    <Modal
        isOpen={props.seekingConfirmation}
        contentLabel="Selected Option"
        onRequestClose={props.cancelConfirmation}
        closeTimeoutMS={200}
        className="modal">

        <h3 className='modal__title'>Delete Expense?</h3>
        <p className='modal__body'>Deleting this expense will permanently remove it from your expense list.</p>
        <div className='modal__buttons'>
            <button
                className='button'
                onClick={props.cancelConfirmation}>
                No, Keep Expense
            </button>
            <button
                className='button button--critical'
                onClick={props.handleRemoveExpense}>
                Yes, Delete Expense
            </button>
        </div>

    </Modal>
);

export default OptionModal;