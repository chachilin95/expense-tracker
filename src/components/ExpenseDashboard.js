import React from 'react';
import ExpenseList from './ExpenseList';

export default class ExpenseDashboard extends React.Component {
    render() {
        return (
            <div>
                <ExpenseList/>
            </div>
        );
    }
}