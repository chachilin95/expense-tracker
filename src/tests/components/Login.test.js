import React from 'react';
import { shallow } from 'enzyme';

import { Login } from '../../components/Login';

let startLoginSpy, wrapper;
beforeEach(() => {
    startLoginSpy = jest.fn();
    wrapper = shallow(<Login startLogin={startLoginSpy}/>);
});

test('Should render Login', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLoginSpy).toHaveBeenCalled();
});