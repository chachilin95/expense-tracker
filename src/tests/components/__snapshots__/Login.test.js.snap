// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
